### PR TITLE
feat(inline-loading): introduce error state

### LINKED ISSUE #3760: Error icon needed for inline loading
### Summary

In the carbon documentation there is an error icon for inline loading - https://www.carbondesignsystem.com/components/inline-loading/usage#error.  

image

However this is not available.  I'm looking to use this for our React UI.

### Justification

We are using this as part of our IBM application, and this guides the user in the correct way.

### Desired UX and success metrics

For the unhappy path of loading, we need to be able to show that there is an error

### "Must have" functionality

There needs to be an error icon for the inline loading

### Specific timeline issues / requests

This is related to our September release, so it would be ideal to have it by the 26th of August, 2019

### Available extra resources

With our tight deadline we are not able to develop this, but we are able to test and get a designer to review.

### PR DESCRIPTION
**Pull request by asudoh**
This change introduces a new state (error) to the inline loading component, in addition to active loading state and finished (complete) state we have had.

Given the inline loading state is no longer binary, `success` prop in React one has been deprecated and replaced with new `status` prop, which takes `inactive`, `active`, `finished` and `error`.

Fixes #3760.

#### Changelog

**New**

- Error state in inline loading
- `state` prop to the React variant, replacing `success` prop

#### Testing / Reviewing

Testing should make sure our inline loading component is not broken.

### CODE REVIEW

**Review by laurenmrice — CHANGES_REQUESTED** [review]
The success icon need to be changed to the spec below.
Inline loader - White theme Copy 2

**Comment by asudoh** [comment]
@laurenmrice Thank you for reviewing! Fixed the icon.

File: packages/react/src/components/InlineLoading/InlineLoading.js [hunk 1]
```diff
@@ -24,39 +27,48 @@ export default function InlineLoading({
...
-    return (
-      <Loading small description={iconDescription} withOverlay={false} active />
-    );
+    return undefined;
   };
   const loadingText = (
     <p className={`${prefix}--inline-loading__text`}>{description}</p>
   );
+  const loading = getLoading();
+  const loadingAnimation = !loading ? (
```
Comment (emyarod) [inline packages/react/src/components/InlineLoading/InlineLoading.js]: would this be identical?

```suggestion
  const loadingAnimation = loading && <div className={`${prefix}--inline-loading__animation`}>{loading}</div>
```

**Review by emyarod — APPROVED** [review]
looks good to me!

**Review by laurenmrice — CHANGES_REQUESTED** [review]
Can you make sure the loading icon is 16px  in both Vanilla and React? Looks larger in React, or maybe not the correct loader? 

The checkmark background color needs to be `interactive-04`. 
Screen Shot 2019-08-22 at 11 07 35 AM

**Comment by asudoh** [comment]
Thanks @laurenmrice for another review! The icon size may have been caused by broken Netlify build upon my earlier commit. Fixed and my apologies for this.

Wrt checkmark background, currently the background color (the checkmark) that icon cannot be changed. We are trying the issue at #3443.

**Comment by laurenmrice** [comment]
@asudoh Oh sorry, the background color that I am referring to here is the color that is currently blue 60 (not the checkmark).

**Review by joshblack — APPROVED** [review]

**Comment by asudoh** [comment]
@laurenmrice I see, thank you for clarifying! Updated.

**Review by laurenmrice — CHANGES_REQUESTED** [review]
In Vanilla: We need the error message to say failed, not filed.

In React:
The success icon is hanging down a bit and need to be center with the type. Make sure its calling the right icon and is the same size and the error and loading one
Screen Shot 2019-08-26 at 12 09 36 PM

**Comment by asudoh** [comment]
@laurenmrice Thank you for spotting those - Fixed both.

**Review by laurenmrice — APPROVED** [review]
looks good 👍

**Comment by gotreasa** [comment]
@asudoh Has the change been deployed to http://react.carbondesignsystem.com/?path=/story/inlineloading--inline-loading or is that yet to happen?

**Comment by laurenmrice** [comment]
It looks like it has not updated react yet @gotreasa

**Comment by asudoh** [comment]
Yeah... You can see it at https://deploy-preview-3772--carbon-components-react.netlify.com/?path=/story/inlineloading--inline-loading for the time being.

**Pull request merged**